hal/src/nRF52840/core_hal.c

```diff
@@ -304,9 +308,6 @@ void HAL_Core_Config(void) {
     else {
         // Set the heap end to the stack start to make most use of the SRAM.
         malloc_set_heap_end(&_Stack_Init);
-
-        // Try copying the embedded user image to user application address
-        user_update();
```

Comment:
Any reason not to call `user_update_if_needed()` here? It looks like we'd avoid checking the module's integrity twice on every boot in this case